File: Companies/VMware/PalindromicSubstrings.cpp
#include <iostream>
#include <string>

using namespace std;

class PalindromicSubstrings
{
public:
  int countSubstrings(string s)
  {
    int re{0}, len = s.size();
    for (int i = 0; i < len; i++)
    {
      for (int l = i, r = i; l >= 0 && r < len && s[l] == s[r]; l--, r++)
      {
        re++;
      }
      for (int l = i, r = i + 1; l >= 0 && r < len && s[l] == s[r]; l--, r++)
      {
        re++;
      }
    }
    return re;
  }
};

File: Companies/Amazon/PalindromicSubstrings.java
package Companies.Amazon;

/**
 * @see LongestPalindromicSubstring
 */
public class PalindromicSubstrings {
    public int countSubstrings(String s) {
        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            count += helper(s, i, i); // odd length
            count += helper(s, i, i+1); // even length
        }
        return count;
    }

    private int helper(String s, int l, int r) {
        int count = 0;
        while (l >= 0 && r < s.length() && (s.charAt(l) == s.charAt(r))) {
            l--;
            r++;
            count++;
        }
        return count;
    }
}
